frame/assets/src/functions.rs

```diff
@@ -554,7 +643,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			ensure!(details.approvals <= witness.approvals, Error::<T, I>::BadWitness);
 
 			for (who, v) in Account::<T, I>::drain_prefix(id) {
-				Self::dead_account(id, &who, &mut details, v.sufficient);
+				let _ = Self::dead_account(id, &who, &mut details, &v.reason);
```

Comment:
Emit a log event if this fails?

Reply:
actually I think we need more than a log, as said here https://github.com/paritytech/substrate/pull/10382#discussion_r758935494
I think any account who is not removed here, because they have their existence due to deposit held will not be able to get the deposit back. Because the asset is removed so `refund` will fail

Reply:
Added a comment.